muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -130,26 +153,59 @@ private NoteRestModel toRestModel(Note note) {
   @GET
   @Path("/owner/{OWNER}")
   @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
-  public Response getNotesByOwner(@PathParam("OWNER") Long owner) {
-    
-    List<Note> notes = notesController.listByOwner(owner);
+  public Response getNotesByOwner(@PathParam("OWNER") Long owner, @QueryParam("listArchived") @DefaultValue ("false") Boolean listArchived) {
+
+    List<Note> notes = notesController.listByOwner(owner, listArchived);
     List<NoteRestModel> notesList = new ArrayList<NoteRestModel>();
+    OffsetDateTime inLastTwoWeeks = OffsetDateTime.now().minusDays(NOTES_FROM_THE_TIME);
     
     for (Note note : notes) {
       NoteRestModel noteRest = toRestModel(note);
       
-      notesList.add(noteRest);
+      UserEntity creator = userEntityController.findUserEntityById(note.getCreator());
+      EnvironmentRoleArchetype creatorUserRole = getUserRoleArchetype(creator.defaultSchoolDataIdentifier());
+      EnvironmentRoleArchetype loggedUserRole = getUserRoleArchetype(sessionController.getLoggedUser());
+      
+      if (loggedUserRole == null || creatorUserRole == null) {
+        return Response.status(Status.BAD_REQUEST).build();
+      }
+      
+      if (Boolean.TRUE.equals(listArchived)) {
+        if (!note.getLastModified().before(Date.from(inLastTwoWeeks.toInstant()))) {
+          if (loggedUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
```

Comment:
Wouldn't it be shorter to just say...

`if (loggedUserRole.equals(EnvironmentRoleArchetype.STUDENT) || !creatorUserRole.equals(EnvironmentRoleArchetype.STUDENT))`

Same with the else block below.